lib/src/destination.dart

```diff
@@ -100,50 +92,19 @@ class Destination {
 
   bool wasParsed = false;
 
-  bool _isUnsupportedScheme;
+  bool? _isUnsupportedScheme;
 
   Destination(Uri uri)
       : url = uri.removeFragment().toString(),
-        _uri = uri.removeFragment() {
-    _hashCode = url.hashCode;
-  }
-
-  factory Destination.fromMap(Map<String, Object> map) {
```

Comment:
The reason I had this `fromMap` <-> `toMap` arrangement was twofold:

1. Caution. I thought maybe there are environments that can't move non-primitive objects from isolate to isolate. I guess that's not a problem on Win/Linux/Mac desktop, so that's okay.
2. Possible problems with mutability. I remember some headaches with having fatal errors when modifying an object after (or during the time?) it was copied to another isolate. I think it was not the case here.

I like the simplicity, though. Let's try this change, and if we see the problems I list above, hopefully one of us will remember where that might be coming from.

Reply:
Makes sense, thanks for the context. With the introduction of lightweight isolates(https://github.com/dart-lang/sdk/issues/36097), they also expanded and more explicitly outlined what can be sent(https://api.dart.dev/stable/2.18.5/dart-isolate/SendPort/send.html) so I think platform support should be fine now. 

I believe the sent objects are also immediately copied, so I don't think there should be any issues during/after the copy, but I'll be on the look out. There could be further benefits to making more of the messages sent deeply immutable, even if just in terms of following the code and avoiding any of those concerns.